lib/screens/translation_screen.dart

```diff
@@ -0,0 +1,360 @@
+        },
+        child: Container(
+            child: _isRecording ? stopIcon : micIcon,
+            padding: new EdgeInsets.all(10.0),
+            decoration: new BoxDecoration(
+                border: new Border.all(color: darkColor, width: 1.0),
+                borderRadius: new BorderRadius.circular(80.0),
+                color: lightColor,
+                boxShadow: [
+                  new BoxShadow(color: darkColor, blurRadius: 20.0)
+                ])));
+
+    return recorder;
+  }
+
+  Widget _controlsWidget() {
+    final textStyle = TextStyle(
+        fontWeight: FontWeight.bold,
+        fontSize: 18.0,
+        color: Theme.of(context).primaryColorDark);
+
+    final controls = Container(
+        decoration: BoxDecoration(
+          color: Colors.white,
+          borderRadius: BorderRadius.circular(30.0),
+        ),
+        padding:
+            EdgeInsets.symmetric(horizontal: padding, vertical: padding / 2),
+        child:
+            Row(mainAxisAlignment: MainAxisAlignment.spaceBetween, children: [
+          Text(_sourceLanguage ?? "", style: textStyle),
+          _recorderWidget(),
+          Text(_targetLang ?? "", style: textStyle),
+        ]));
+    return Padding(padding: EdgeInsets.all(padding), child: controls);
+  }
+
+  Widget _timerWidget() {
+    return Text("$_recorderTimer", style: TextStyle(fontSize: 30.0));
+  }
+
+  Widget _buildSentencePageWidget(data) {
+    final itemList = PageView.builder(
+      controller: _pageController,
+      itemCount: data.length,
+      itemBuilder: (context, index) {
+        Sentence item = data[index];
+        return _buildSentenceCard(item);
+      },
+      onPageChanged: (index) {
+        setState(() {
+          _currentSentence = data[index];
+          _recorderTimer = "";
+          _progression = index + 1;
+        });
+      },
+    );
+
+    return Container(child: itemList, padding: EdgeInsets.all(16.0));
+  }
+
+  Widget _buildSentenceCard(Sentence phrase) {
+    return Card(
+        elevation: 2.0,
+        child: Container(
+            height: MediaQuery.of(context).size.height / 2,
+            width: MediaQuery.of(context).size.width,
+            padding: EdgeInsets.all(padding),
+            child: Center(
+                child: Text(
+              phrase.text,
+              style: TextStyle(
+                  color: Theme.of(context).accentColor,
+                  letterSpacing: 0.75,
+                  fontSize: 16.0,
+                  fontWeight: FontWeight.bold),
+            ))));
+  }
+
+  Widget _loadSentencesWidget() {
+    return FutureBuilder(
+        future: _fetchAndCacheData(),
+        builder: (context, snapshot) {
+          if (!snapshot.hasError && snapshot.hasData) {
+            _sentences = snapshot.data;
+            return _buildSentencePageWidget(_sentences);
+          } 
+          else {
+            return Center(child: CircularProgressIndicator());
```

Comment:
do we want special treatment for hasError() depending on the type of error? say no internet connection do we just want to show progress bar? This is not super important we can discuss it in next milestones.

Reply:
I noticed some issues when dealing with every type of errors. So for now I will leave this as this. But I will add a TODO there.